src/txHashBuilder.c

```diff
@@ -861,33 +861,33 @@ static void txHashBuilder_assertCanLeaveWithdrawals(tx_hash_builder_t* builder)
 	ASSERT(builder->remainingWithdrawals == 0);
 }
 
-void txHashBuilder_addMetadata(tx_hash_builder_t* builder, const uint8_t* metadataHashBuffer, size_t metadataHashBufferSize)
+void txHashBuilder_addAuxData(tx_hash_builder_t* builder, const uint8_t* auxDataHashBuffer, size_t auxDataHashBufferSize)
 {
 	txHashBuilder_assertCanLeaveWithdrawals(builder);
-	ASSERT(builder->includeMetadata);
+	ASSERT(builder->includeAuxData);
 
 	{
 		BUILDER_APPEND_CBOR(CBOR_TYPE_UNSIGNED, 7);
```

Comment:
The `TX_BODY_KEY_METADATA` constant should probably be used here. And it should also probably be renamed to `TX_BODY_KEY_AUXILIARY_DATA` then. Or maybe even `TX_BODY_KEY_AUXILIARY_DATA_HASH`.